.github/workflows/test.yml

```diff
@@ -111,25 +81,67 @@ jobs:
           - Debug
           - Release
     steps:
-      - name: Checkout with submodules
-        uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
-          submodules: recursive
+          fetch-depth: 0
       - name: Install Winflexbison3
         shell: powershell
         run: choco install winflexbison3 --version 2.5.24.20210105
+      - name: Get latest CMake
+        uses: lukka/get-cmake@latest
       - name: Create build directory
-        run: mkdir -p build-${{ matrix.build_type }}
+        run: mkdir -p build
       - name: Configure
-        working-directory: build-${{ matrix.build_type }}
-        run: cmake -S .. -B . -DQX_BUILD_TESTS=ON
+        working-directory: build
+        run: cmake -S .. -B . -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DQX_BUILD_TESTS=ON
       - name: Build
-        working-directory: build-${{ matrix.build_type }}
+        working-directory: build
         run: cmake --build . --parallel 10 --config ${{ matrix.build_type }}
       - name: Test
-        working-directory: build-${{ matrix.build_type }}
+        working-directory: build
         run: ctest -C ${{ matrix.build_type }} --output-on-failure
 
+  cpp-arm64:
+    name: 'C++ tests (gcc/Linux/ARM64, clang/MacOS/ARM64)'
+    runs-on: [self-hosted, ARM64, "${{ matrix.os }}"]
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - Linux
+          - macOS
+    steps:
+      - if: matrix.os == 'Linux'
+        name: Install flex/bison, and gcc (Linux)
+        run: |
+          sudo apt-get -y update
+          sudo apt-get -y upgrade
+          sudo apt-get -y install flex bison gcc
+        shell: bash
+      - if: matrix.os == 'macOS'
+        name: Install flex/bison, and gcc (MacOS)
+        run: |
+          brew install flex bison gcc
+          echo "$(brew --prefix flex)/bin" >> $GITHUB_PATH
+          echo "$(brew --prefix bison)/bin" >> $GITHUB_PATH
+        shell: bash
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

Comment:
I originally thought this fetch-depth was a nice optimization, but in fact setting it to 0 seems to fetch ALL history for all branches and tags, which is useless right?
```
    # Number of commits to fetch. 0 indicates all history for all branches and tags.
    # Default: 1
    fetch-depth: ''
```

See https://github.com/actions/checkout

I think we only need the files, history is irrelevant no?

Reply:
Yes, again, this is something that I carry on from other projects, and I don't really know exactly why I started using it in the first place. So setting it to its default, yes. Thanks!